Libraries/Opc.Ua.Gds.Server.Common/CertificateGroup.cs

```diff
@@ -319,35 +319,34 @@ string subjectName
             }
 
             DateTime yesterday = DateTime.Today.AddDays(-1);
-            using (X509Certificate2 newCertificate = CertificateFactory.CreateCertificate(subjectName)
+            using (X509Certificate2 newCertificate = await CertificateFactory.CreateCertificate(subjectName)
                 .SetNotBefore(yesterday)
                 .SetLifeTime(Configuration.CACertificateLifetime)
                 .SetHashAlgorithm(X509Utils.GetRSAHashAlgorithmName(Configuration.CACertificateHashSize))
                 .SetCAConstraint()
                 .SetRSAKeySize(Configuration.CACertificateKeySize)
                 .CreateForRSA()
-                .AddToStore(
+                .AddToStoreAsync(
                     AuthoritiesStoreType,
-                    AuthoritiesStorePath))
+                    AuthoritiesStorePath).ConfigureAwait(false))
             {
 
                 // save only public key
                 Certificate = new X509Certificate2(newCertificate.RawData);
 
                 // initialize revocation list
-                Task<X509CRL> crlTask = RevokeCertificateAsync(AuthoritiesStorePath, newCertificate, null);
+                X509CRL crl = await RevokeCertificateAsync(AuthoritiesStorePath, newCertificate, null).ConfigureAwait(false);
 
                 //Update TrustedList Store
-                await crlTask.ContinueWith((_) =>
-                    UpdateAuthorityCertInCertificateStore(Configuration.TrustedListPath)
-                    , TaskContinuationOptions.OnlyOnRanToCompletion).ConfigureAwait(false);
-
-                //Update TrustedIssuerCertificates Store
-                if (!string.IsNullOrEmpty(TrustedIssuerCertificatesStorePath))
+                if (crl != null)
                 {
-                    await crlTask.ContinueWith((_) =>
-                        UpdateAuthorityCertInCertificateStore(TrustedIssuerCertificatesStorePath)
-                        , TaskContinuationOptions.OnlyOnRanToCompletion).ConfigureAwait(false);
+                    await UpdateAuthorityCertInCertificateStore(Configuration.TrustedListPath).ConfigureAwait(false);
```

Comment:
the 'old' way update is not called with await, so it may just run in the wild? Tried to fix it.